### DISCUSSION TITLE
Confusion about lin_l and lin_r methods in SAGEConv class

### DISCUSSION THREAD

**Discussion by YiheWang** [pyg-team/pytorch_geometric#5789] [Q&A] [answered]
Hi, I am confused about lin_l and lin_r methods in SAGEConv class. 
![image](https://user-images.githubusercontent.com/36209840/197034072-e4b55b81-4743-4f0d-975b-136e18655cee.png)

Which one is W1, and which one is W2 respectively? 
![4d3781f07776f825fc8b2c37faeeb00](https://user-images.githubusercontent.com/36209840/197028211-742dcf3e-3de9-4cb4-95ab-7c2a0f20a2ad.jpg)

Besides, I found that W1 and W2 are actually the same in the original paper if apply the mean aggregator. They concatenate x_i and the mean of x_j before dotting with W. I am wondering if you follow the paper's idea or do some modifications? 
![image](https://user-images.githubusercontent.com/36209840/197034502-577687c1-7107-4eb9-85b0-239480a1c829.png)

Thanks in advance for your help!

**Comment by rusty1s** [comment, marked as answer]
`lin_l` refers to W_2, transforming the source/left features (sorry for poor wording), and `lin_r` refers to W_1. Note that transformation + summation is identical to concatenation + transformation. The former is more memory-efficient, that's why we use it here.

**Reply by YiheWang** [reply]
>Note that transformation + summation is identical to concatenation + transformation.

I understand now. Thanks for your help!
